cprover-cpp-api/src/api.h

```diff
@@ -0,0 +1,36 @@
+#ifndef CPROVER_API_API_H
+#define CPROVER_API_API_H
+
+#include <memory>
+#include <vector>
+#include <string>
+
+// Forward declaration of API dependencies
+class goto_modelt;
+class message_handlert;
+class optionst;
+
+#include "options.h"
+
+// An object in the pattern of Session Facade - owning all of the memory
+// the API is using and being responsible for the management of that.
+struct cprover_api
+{
+  // Initialising constructor
+  cprover_api();
+  explicit cprover_api(const api_optionst &options);
+  ~cprover_api(); // default constructed in the .cpp file
```

Comment:
This would read better to me if the word `constructed` were to be replaced with `definition`.